dotnet/SimpleSauce/SauceSession.cs

```diff
@@ -9,26 +9,38 @@ public class SauceSession
     {
         public ChromeOptions ChromeOptions { get; private set; }
         public DataCenter DataCenter { get; set; } = DataCenter.UsWest;
+        public SauceOptions Options { get; }
 
-        private readonly IRemoteDriver _remoteDriverManager;
-        private readonly SauceOptions options;
+        public ISauceRemoteDriver DriverImplementation { get; set; }
 
         public SauceSession()
         {
-            _remoteDriverManager = new ConcreteRemoteWebDriver();
+            DriverImplementation = new SauceDriver();
+            Options = new SauceOptions();
         }
-        public SauceSession(IRemoteDriver driverManager)
+        public SauceSession(ISauceRemoteDriver driver)
         {
-            _remoteDriverManager = driverManager;
+            DriverImplementation = driver;
+            Options = new SauceOptions();
         }
 
         public SauceSession(SauceOptions options)
         {
-            this.options = options;
+            this.Options = options;
+            DriverImplementation = new SauceDriver();
+        }
+
+        public SauceSession(SauceOptions options, ISauceRemoteDriver driver)
+        {
+            this.Options = options;
+            DriverImplementation = driver;
         }
 
         public IWebDriver Start()
         {
+            if (Options.EdgeOptions != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4510198)